ingest-router/src/project_config/handler.rs

```diff
@@ -100,32 +109,35 @@ impl Handler<ProjectConfigsRequest, ProjectConfigsResponse> for ProjectConfigsHa
 
         let mut merged = ProjectConfigsResponse::new();
 
-        // Add pending keys from split phase
-        merged.pending_keys.extend(pending_from_split);
-
-        // Results are provided pre-sorted by cell priority (highest first)
-        // The executor ensures results are ordered so we can use the first successful response
-        // for extra_fields and headers.
-        // Failed cells are handled by the executor adding their keys to pending_from_split.
-        let mut iter = results.into_iter().flatten();
+        // Downcast metadata to our specific type
+        if let Ok(project_metadata) = metadata.downcast::<ProjectConfigsMetadata>() {
+            merged.pending_keys.extend(*project_metadata);
+        }
```

Comment:
this is kind of the downside of this approach -- it's probably worth attempting to write a macro or enum here to avoid this need to manually downcast to the real type.